dlt/pipeline/track.py

```diff
@@ -61,21 +64,23 @@ def _get_step_elapsed(step: PipelineStepTrace) -> str:
 
 def on_start_trace(trace: PipelineTrace, step: TPipelineStep, pipeline: SupportsPipeline) -> None:
     if pipeline.runtime_config.sentry_dsn:
-        # https://getsentry.github.io/sentry-python/api.html#sentry_sdk.Hub.capture_event
-        # print(f"START SENTRY TX: {trace.transaction_id} SCOPE: {Hub.current.scope}")
-        transaction = Hub.current.start_transaction(name=step, op=step)
-        _add_sentry_tags(transaction, pipeline)
-        transaction.__enter__()
+        # print(f"START SENTRY TX: {trace.transaction_id} SCOPE: {Hub.current.scope}"
+        global _current_transactions
+        transaction = Scope.get_current_scope().start_transaction(name=step, op=step)
+        if isinstance(transaction, Transaction):
+            _add_sentry_tags(transaction, pipeline)
+            _current_transactions.append(transaction.__enter__())
 
 
 def on_start_trace_step(
     trace: PipelineTrace, step: TPipelineStep, pipeline: SupportsPipeline
 ) -> None:
     if pipeline.runtime_config.sentry_dsn:
         # print(f"START SENTRY SPAN {trace.transaction_id}:{trace_step.span_id} SCOPE: {Hub.current.scope}")
-        span = Hub.current.scope.span.start_child(description=step, op=step).__enter__()
-        span.op = step
+        global _current_spans
+        span = Scope.get_current_scope().start_span(description=step, op=step)
```

Comment:
I did not find a way to get the current span, so I'm using a fifo list for this now.